packages/grafana-ui/src/components/Combobox/Combobox.tsx

```diff
@@ -144,34 +157,62 @@ export const Combobox = <T extends string | number>({
     defaultHighlightedIndex: selectedItemIndex ?? 0,
     scrollIntoView: () => {},
     onInputValueChange: ({ inputValue }) => {
-      const filteredItems = options.filter(itemFilter(inputValue));
-      if (createCustomValue && inputValue && filteredItems.findIndex((opt) => opt.label === inputValue) === -1) {
-        const customValueOption: ComboboxOption<T> = {
-          label: inputValue,
-          // @ts-ignore Type casting needed to make this work when T is a number
-          value: inputValue as unknown as T,
-          description: t('combobox.custom-value.create', 'Create custom value'),
-        };
-
-        setItems([...filteredItems, customValueOption]);
+      const customValueOption =
+        createCustomValue &&
+        inputValue &&
+        items.findIndex((opt) => opt.label === inputValue || opt.value === inputValue) === -1
+          ? {
+              // Type casting needed to make this work when T is a number
```

Comment:
The generic type would suggest that the combobox can be either only all numbers or all strings but then this cast does not make sense because value really is a string and we are breaking the expected constraint of all values being numbers if using `Combobox<number>`. That could possibly break other parts like `onChange` which will now get either number or string.

Reply:
Yeah - we struggled a lot with the types here, and this is just a shitty compromise.

I agree - we need to fix these up.